api/src/domain/user/updateOperatingPhase.test.ts

```diff
@@ -84,20 +118,40 @@ describe("updateOperatingPhase", () => {
           operatingPhase: "NEEDS_TO_FORM",
           legalStructureId: "general-partnership",
         }),
-        taskProgress: { [formationTaskId]: "IN_PROGRESS" },
+        taskProgress: { [formationTaskId]: "IN_PROGRESS", [businessStructureTaskId]: "COMPLETED" },
```

Comment:
I think this outcome is depenedent on the Tax Task, not the formation task (since this is for trade name)

even if the formation task was complete here, the outcome would be the same. 
I think the test should make that explicit in the test name (that it's the tax task being incomplete that makes it needs-to-register)